### PR TITLE
[6.13.z] autodetect and add prt tests for dependabot PR's

### PR DESCRIPTION
**Pull request by Satellite-QE**
Cherrypick of PR: https://github.com/SatelliteQE/robottelo/pull/11301

### PR Description 

-  GitHub action to detect if a pull request was created by Dependabot.
- Based on the dependencies being updated by Dependabot, the action fetches tests from a `tests.yaml `file.
- The action makes PRT (pull request testing) comments that will eventually trigger the tests to run in a continuous integration (CI) system.
- Once the tests have completed running, the results are provided in the form of a status update on the GitHub pull request.

Overall, your action helps to automate the testing process for Dependabot PRs by automatically detecting what tests need to be run and triggering them in the CI system. This will ensure that changes made by Dependabot are thoroughly tested before being merged into the codebase.

This is verified on my own fork here : https://github.com/omkarkhatavkar/robottelo/pull/72

### CODE REVIEW

**Review by ogajduse — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ updates:
     labels:
       - "CherryPick"
       - "dependencies"
+      - "6.13.z"
```
Comment (Gauravtalreja1) [inline .github/dependabot.yml:15]: Can we add 6.14.z here too?

File: .github/dependency_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+broker[docker]: "tests/foreman/ -k 'test_host_registration_end_to_end or test_positive_erratum_applicability or test_positive_upload_content'"
+deepdiff: "tests/foreman/endtoend/test_api_endtoend.py -k 'test_positive_get_links'"
+dynaconf[vault]: "tests/foreman/api/test_ldapauthsource.py -k 'test_positive_endtoend'"
+manifester: "tests/foreman/api/test_subscription.py -k 'test_positive_create_after_refresh'"
+navmazing: "tests/foreman/ui/test_repository.py -k 'test_positive_create_as_non_admin_user'"
+pyotp: "tests/foreman/ui/test_ldap_authentication.py -k 'test_positive_login_user_password_otp'"
+pytest-xdist: "tests/foreman/ -n 3 -m 'build_sanity'"
```
Comment (Gauravtalreja1) [inline .github/dependency_tests.yaml:7]: I think it'd be better to run this with 2 workers, wdyt?
```suggestion
pytest-xdist: "tests/foreman/ -n 2 -m 'build_sanity'"
```

File: .github/workflows/auto_cherry_pick_merge.yaml [hunk 3]
```diff
@@ -48,54 +48,23 @@ jobs:
...
-            status=$(get_status)
-            echo "Robottelo-Runner : $status"
-            counter=$((counter+1))
-          done
+      - name: Fetch the PRT status
+        id: outcome
+        uses: omkarkhatavkar/wait-for-status-checks@main
+        with:
+          ref: ${{ github.head_ref }}
+          context: 'Robottelo Runner'
```
Comment (Gauravtalreja1) [inline .github/workflows/auto_cherry_pick_merge.yaml:56]:
```suggestion
          context: 'Robottelo-Runner'
```

File: .github/dependency_tests.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+pytest: "tests/foreman/ -m 'build_sanity'"
+python-box: "tests/foreman/cli/test_webhook.py -k 'test_positive_end_to_end'"
+requests: "tests/foreman/cli/test_repository.py -k 'test_file_repo_contains_only_newer_file'"
+wait-for: "tests/foreman/api/test_reporttemplates.py -k 'test_positive_schedule_entitlements_report'"
+wrapanapi: "tests/foreman/longrun/test_provisioning_computeresource.py -k 'test_positive_provision_vmware_with_host_group'"
```
Comment (Gauravtalreja1) [inline .github/dependency_tests.yaml:12]: VMWare tests usually fail due to unavailability of vlan_networking so would suggest better to use GCP or some other tests

File: .github/dependency_tests.yaml (hunk 2 above)
Comment (omkarkhatavkar) [reply .github/dependency_tests.yaml:7]: That's fine I guess it has to be more than 1.

File: .github/dependabot.yml (hunk 1 above)
Comment (omkarkhatavkar) [reply .github/dependabot.yml:15]: I think we might not require this here. because we do not want the PR's for this non-default(non master) branches

Comment (omkarkhatavkar) [reply .github/dependabot.yml:15]: I will remove these branches from here and add 6.14 in the master

Comment (omkarkhatavkar) [reply .github/dependabot.yml:15]: That needs to be added in the master

File: .github/dependency_tests.yaml (hunk 2 above)
Comment (Gauravtalreja1) [reply .github/dependency_tests.yaml:7]: yeah, but less the better as it will checkout this number of satellites

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**